modules/imgproc/src/imgwarp.cpp

```diff
@@ -5960,6 +5964,8 @@ void cv::warpAffine( InputArray _src, OutputArray _dst,
                      InputArray _M0, Size dsize,
                      int flags, int borderType, const Scalar& borderValue )
 {
+    CV_Assert( _src.channels() <= 4 || !(flags & (INTER_LANCZOS4 | INTER_CUBIC) ) );
+
     CV_INSTRUMENT_REGION()
```

Comment:
`CV_INSTRUMENT_REGION()` should be the first line of function.

`INTER_LANCZOS4` and `INTER_CUBIC` are not a binary flags in general, so such expressions are not seems valid.
Condition should be like this:
```
int interpolation = flags & INTER_MAX;
... (interpolation == INTER_CUBIC || interpolation == INTER_LANCZOS4)
```